### PR TITLE
feat: parallel testing

### PR DESCRIPTION
**Pull request by chiefbiiko**
Defaults to serial execution.
Test parallel with `runTests({ parallel: true })`.
Run `deno ./testing/bench.ts` for a simple comparison.

### CODE REVIEW

File: testing/mod.ts [hunk 1]
```diff
@@ -245,55 +245,181 @@ function green_ok() {
...
-    const { fn, name } = tests[i];
-    let result = green_ok();
+export async function runIfMain(meta: ImportMeta, opts?: RunOptions) {
+  if (meta.main) {
+    return runTests(opts);
+  }
+}
+
+export interface RunOptions {
+  serial?: boolean;
```
Comment (j-f1) [inline testing/mod.ts]: Why not pass `parallel: true` instead of `serial: false`?

**Comment by zekth** [comment]
Would it be better to set a value `{ concurrent: true }` instead of `{ serial: false }`. In term of common usage in testing it has more sense to set a flag of the execution?

File: testing/mod.ts (hunk 1 above)
Comment (ry) [reply testing/mod.ts]: I agree, this seems more intuitive.

File: testing/mod.ts [hunk 2]
```diff
@@ -245,55 +245,181 @@ function green_ok() {
...
-  for (let i = 0; i < tests.length; i++) {
-    const { fn, name } = tests[i];
-    let result = green_ok();
+export async function runIfMain(meta: ImportMeta, opts?: RunOptions) {
+  if (meta.main) {
+    return runTests(opts);
+  }
+}
+
+export interface RunOptions {
```
Comment (ry) [inline testing/mod.ts]: Add jsdocs here explaining the defaults.

File: testing/mod.ts [hunk 3]
```diff
@@ -245,55 +245,181 @@ function green_ok() {
...
 
-export async function runTests() {
-  let passed = 0;
-  let failed = 0;
-
-  console.log("running", tests.length, "tests");
-  for (let i = 0; i < tests.length; i++) {
-    const { fn, name } = tests[i];
-    let result = green_ok();
+export async function runIfMain(meta: ImportMeta, opts?: RunOptions) {
```
Comment (ry) [inline testing/mod.ts]: Add jsdocs here explaining the defaults.

File: testing/mod.ts (hunk 1 above)
Comment (chiefbiiko) [reply testing/mod.ts]: Thanks for the feedback, now `parallel`.

File: testing/mod.ts (hunk 3 above)
Comment (chiefbiiko) [reply testing/mod.ts]: updated

**Pull request merged**